pkg/controller/projectreference/projectreference_controller.go

```diff
@@ -72,34 +88,176 @@ func (r *ReconcileProjectReference) Reconcile(request reconcile.Request) (reconc
+			// TODO: add condition here SupportedRegion = false to give more information on the error state
+			reqLogger.Error(err, "Region not supported")
+			projectReference.Status.State = gcpv1alpha1.ProjectReferenceStatusError
+			err := r.client.Status().Update(context.TODO(), projectReference)
+			if err != nil {
+				reqLogger.Error(err, "Error updating ProjectReference Status")
+				return r.requeueOnErr(err)
+			}
+			return r.doNotRequeue()
+		}
+
+		reqLogger.Info(fmt.Sprintf("Setting ProjectReferenceStatus %s", gcpv1alpha1.ProjectReferenceStatusCreating))
+		// passed requirementes check set to creating
+		projectReference.Status.State = gcpv1alpha1.ProjectReferenceStatusCreating
+		err = r.client.Status().Update(context.TODO(), projectReference)
+		if err != nil {
+			reqLogger.Error(err, "Error updating ProjectReference Status")
+			return r.requeueOnErr(err)
+		}
+	}
+
+	if projectReference.Spec.GCPProjectID == "" {
+		reqLogger.Info("Creating ProjectID in ProjectReference CR")
+		err := adapter.updateProjectID()
+		if err != nil {
+			reqLogger.Error(err, "Could not update ProjectID in Project Reference CR")
+			return r.requeueOnErr(err)
+		}
+		return r.requeue()
+	}
+
+	//TODO() add configmap to ReferenceAdapter
+	parentFolderID, err := util.GetGCPParentFolderFromConfigMap(r.client, operatorNamespace, orgGcpConfigMap)
+	if err != nil {
+		reqLogger.Error(err, "could not get orgParentFolderID from the ConfigMap:", orgGcpConfigMap, "Operator Namespace", operatorNamespace)
+		return r.requeueOnErr(err)
+	}
+
+	reqLogger.Info("Configuring Project")
+	err = adapter.createProject(parentFolderID)
+	if err != nil {
+		if err == operrors.ErrInactiveProject {
+			log.Error(err, "Unrecoverable Error")
+			projectReference.Status.State = gcpv1alpha1.ProjectReferenceStatusError
+			err := r.client.Status().Update(context.TODO(), projectReference)
+			if err != nil {
+				reqLogger.Error(err, "Error updating ProjectReference Status")
+				return r.requeueOnErr(err)
+			}
+		}
+		reqLogger.Error(err, "Could not create ProjectID")
+		return r.requeueOnErr(err)
+	}
+
+	reqLogger.Info("Configuring APIS")
+	// TODO() Set condition billing has been created and skip that this step if condition is true
+	err = adapter.configureAPIS()
+	if err != nil {
+		reqLogger.Error(err, "Error configuring APIS")
+		return r.requeueAfter(5*time.Second, err)
+	}
+
+	reqLogger.Info("Configuring Service Account")
+	err = adapter.configureSeriveAccount()
+	if err != nil {
+		reqLogger.Error(err, "Error configuring service account")
+		return r.requeueAfter(5*time.Second, err)
+	}
+
+	reqLogger.Info("Creating Credentials")
+	err = adapter.createCredentials()
+	if err != nil {
+		reqLogger.Error(err, "Error creating credentials")
+		return r.requeueAfter(5*time.Second, err)
+	}
+
+	log.Info("Setting Status on projectReference")
+	projectReference.Status.State = gcpv1alpha1.ProjectReferenceStatusReady
+	err = r.client.Status().Update(context.TODO(), projectReference)
```

Comment:
Not sure what happens on a call of `Update()` if nothing changed, hopefully it doesn't result in reconciling the ProjctClaim again. Maybe it makes sense to explicitly check this before?

Reply:
@NautiluX This is calling Update specifically on the Status since we are updating it to ready.

Reply:
and it does requeue. If you run the code you will see that on the next reconcile it will be caught at line 138 and then it will update the ProjectClaim status to ready as well.